wagtail/admin/menu.py

```diff
@@ -44,17 +44,20 @@ def render_component(self, request):
 class DismissibleMenuItemMixin:
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
-        self.attrs["data-wagtail-dismissible-id"] = self.name
+        self.attrs["data-controller"] = "w-dismissible"
```

Comment:
I actually think we do not need the controller OR the action on the JSX.
The React component does this same work for us (hence the back and forth on the main issue).

The reason we wanted to adopt the `value` with the `data-w-dismissible-id-value` is for consistency.

maybe... what if we keep the controller and remove the action.

keep the data-w-dismissible-dismissed-class - the JSX does not appear to use it but makes sense to keep it for consistency.

Reply:
We might not need the action
But I'm sure we do need to set the dismissed-value
because without it, 
```
function getInitialDismissibleState(menuItems: MenuItemDefinition[]) {
  const result: Record<string, boolean> = {};

  walkDismissibleMenuItems(menuItems, (menuItem) => {
    result[menuItem.attrs['data-w-dismissible-id-value']] =
      'data-w-dismissible-dismissed-value' in menuItem.attrs;
  });

  return result;
}
```
This would be empty which means 

![image](https://user-images.githubusercontent.com/38161296/232000320-0be1bb18-a561-4e32-b709-76909a11fd83.png)

would never show the notification icon

So I can leave the id right and then remove the action

Reply:
Good call. Thanks for digging into it.